app/controllers/api/v2/vcs_clone_controller.rb

```diff
@@ -0,0 +1,150 @@
+        param :ref, String, :desc => N_('Branch / Tag / Commit reference'), :required => true
+      end
+      param :smart_proxy, Array, N_('SmartProxy the role should get installed to')
+      error 400, :desc => N_('Parameter unfulfilled')
+      def update_role
+        payload = verify_task_parameters(params)
+        start_vcs_task(payload, :update)
+      end
+
+      api :DELETE, '/smart_proxies/:proxy_name/ansible/vcs_clone/roles/:role_name',
+          N_('Launches a task to delete the provided role')
+      formats ['json']
+      param :role_name, String, :desc => N_('Name of the role that should be deleted')
+      param :smart_proxy, Array, N_('SmartProxy the role should get deleted from')
+      error 400, :desc => N_('Parameter unfulfilled')
+      def delete_role
+        payload = params.require(:role_name)
+        start_vcs_task(payload, :delete)
+      end
+
+      private
+
+      def set_proxy_api
+        unless params[:id]
+          msg = _('Smart proxy id is required')
+          return render_error('custom_error', :status => :unprocessable_entity, :locals => { :message => msg })
+        end
+        ansible_proxy = SmartProxy.find_by(name: params[:id])
+        if ansible_proxy.nil?
+          msg = _('Smart proxy does not exist')
+          return render_error('custom_error', :status => :bad_request, :locals => { :message => msg })
+        else unless ansible_proxy.has_capability?('Ansible', 'vcs_clone')
+               msg = _('Smart proxy does not have foreman_ansible installed / is not capable of cloning from VCS')
+               return render_error('custom_error', :status => :bad_request, :locals => { :message => msg })
+             end
+        end
+        @proxy = ansible_proxy
+        @proxy_api = find_proxy_api(ansible_proxy)
+      end
+
+      def verify_task_parameters(params)
+        payload = params.require(:vcs_clone).
+                  permit(
+                    repo_info: [
+                      :vcs_url,
+                      :name,
+                      :ref
+                    ]
+                  ).to_h
+        %w[vcs_url name ref].each do |param|
+          raise ActionController::ParameterMissing.new(param) unless payload['repo_info'].key?(param)
+        end
+        payload
+      end
+
+      def start_vcs_task(op_info, operation)
+        case operation
+        when :update
+          job = UpdateAnsibleRole.perform_later(op_info, @proxy)
+        when :install
+          job = CloneAnsibleRole.perform_later(op_info, @proxy)
+        when :delete
+          job = DeleteAnsibleRole.perform_later(op_info, @proxy)
+        else
+          raise Foreman::Exception.new(N_('Unsupported operation'))
+        end
+
+        task = ForemanTasks::Task.find_by(external_id: job.provider_job_id)
+
+        render json: {
+          task: task
+        }, status: :ok
+      rescue Foreman::Exception
+        head :internal_server_error
+      end
+
+      private
```

Comment:
There's already a `private` statement on line 80.
```suggestion
```

Reply:
The whole block was useless. Not sure, how I missed that...